src/Illuminate/Foundation/Http/Middleware/EnsureRequestIsSecure.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace Illuminate\Foundation\Http\Middleware;
+
+use Closure;
+
+class EnsureRequestIsSecure
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (! $request->secure() && config('app.https')) {
```

Comment:
This check should be switched I think, since it's not necessary to check if the request is secure when that's not wanted in the config?

```php
if (config('app.https') && !$request->secure()) {
  // ...
}
```